[BUG] Cannot add new Spell properties via World Script/Module - 5.1.10

**Issue**
Adding a custom property to Spells via a World Script/Module appears to succeed, but attempting to toggle the property on any spell throws errors. If the property is enabled and the world is reloaded, the Actor sheet becomes inaccessible.

Example Script:
```js
Hooks.once("init", () => {
  CONFIG.DND5E.itemProperties.test = {
    label: "Test"
  };
  CONFIG.DND5E.validProperties.spell.add("test");
});
```

Error on Toggle:

```
Error: Failed data preparation for Actor.ZmpSKnvlUoSfTYTc.Item.P97npemu7j70IZAQ. Iterable yielded undefined which is not a string
    at Hooks.onError (foundry.mjs:23563:24)
    at Item5e._safePrepareData (foundry.mjs:32665:17)
    at Actor5e.prepareEmbeddedDocuments (foundry.mjs:32726:13)
    at Actor5e.prepareEmbeddedDocuments (foundry.mjs:41981:11)
    at Actor5e.prepareEmbeddedDocuments (actor.mjs:265:11)
    at Actor5e.prepareData (foundry.mjs:32702:12)
    at Actor5e.prepareData (foundry.mjs:41965:11)
    at Actor5e.prepareData (flags.mjs:23:13)
    at Actor5e.prepareData (actor.mjs:243:11)
    at Actor5e._safePrepareData (foundry.mjs:32663:14)Caused by: TypeError: Iterable yielded undefined which is not a string
    at ListFormat.format (<anonymous>)
    at SpellData.prepareDerivedData (spell.mjs:257:77)
    at Item5e.prepareData (foundry.mjs:32703:37)
    at Item5e.prepareData (flags.mjs:23:13)
    at Item5e._safePrepareData (foundry.mjs:32663:14)
    at Actor5e.prepareEmbeddedDocuments (foundry.mjs:32726:13)
    at Actor5e.prepareEmbeddedDocuments (foundry.mjs:41981:11)
    at Actor5e.prepareEmbeddedDocuments (actor.mjs:265:11)
    at Actor5e.prepareData (foundry.mjs:32702:12)
    at Actor5e.prepareData (foundry.mjs:41965:11)
```

And Error on Reload:
```
foundry.mjs:23564 Error: Failed data preparation for Actor.ZmpSKnvlUoSfTYTc. Cannot read properties of undefined (reading 'byType')
    at Hooks.onError (foundry.mjs:23563:24)
    at Actor5e._safePrepareData (foundry.mjs:32665:17)
    at Game.initializeDocuments (foundry.mjs:172380:18)
    at Game.setupGame (foundry.mjs:172205:10)
    at async #initializeGameView (foundry.mjs:173715:5)
    at async Game.initialize (foundry.mjs:172147:5)
    at async window.addEventListener.once (foundry.mjs:184183:3)Caused by: TypeError: Cannot read properties of undefined (reading 'byType')
    at get scaleValues (item.mjs:360:28)
    at actor-data-model.mjs:62:32
    at EmbeddedCollection.reduce (foundry.mjs:5806:21)
    at CharacterData._prepareScaleValues (actor-data-model.mjs:61:36)
    at CharacterData.prepareEmbeddedData (actor-data-model.mjs:50:10)
    at CharacterData.prepareEmbeddedData (character.mjs:241:11)
    at Actor5e.applyActiveEffects (actor.mjs:273:77)
    at Actor5e.prepareEmbeddedDocuments (foundry.mjs:41982:10)
    at Actor5e.prepareEmbeddedDocuments (actor.mjs
```

Adding the property to feats or equipment works without any problems.


Tested in a new World without Modules
- Foundry VTT V13 350
- DnD 5e v5.1.10

--- a/module/data/item/spell.mjs
+++ b/module/data/item/spell.mjs
@@ -375,9 +375,17 @@ export default class SpellData extends ItemDataModel.mixin(ActivitiesTemplate, I
       const config = this.validProperties.has(c) ? CONFIG.DND5E.itemProperties[c] : null;
       if ( !config ) return obj;
       const { abbreviation: abbr, label, icon } = config;
-      obj.all.push({ abbr, icon, tag: config.isTag });
-      if ( config.isTag ) obj.tags.push(label);
-      else obj.vsm.push(abbr);
+      // Only add properties to display arrays if they have displayable content
+      if ( config.isTag ) {
+        // Tag properties: add to tags if has label
+        if ( label ) obj.tags.push(label);
+        if ( abbr || icon ) obj.all.push({ abbr, icon, tag: true });
+      } else if ( abbr ) {
+        // VSM properties: only add if has abbreviation
+        obj.vsm.push(abbr);
+        obj.all.push({ abbr, icon, tag: false });
+      }
+      // Properties with neither abbreviation nor isTag are silently ignored for display
       return obj;
     }, { all: [], vsm: [], tags: [] });
     labels.components.vsm = game.i18n.getListFormatter({ style: "narrow" }).format(labels.components.vsm);

